src/core/Header/HeaderTitle.tsx

```diff
@@ -0,0 +1,45 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+import '@itwin/itwinui-css/css/header.css';
+
+import cx from 'classnames';
+import React from 'react';
+
+import { useTheme } from '../utils/hooks/useTheme';
+
+export type HeaderTitleProps = {
+  /**
+   * Logo shown before the main content.
+   */
+  logo: JSX.Element;
+} & React.ButtonHTMLAttributes<HTMLDivElement>;
+
+/**
+ * Header Title section
+ * @example
+ * <HeaderTitle logo="<SvgMicrostation />">Microstation</HeaderTitle>
+ * <HeaderTitle logo="<SvgItwin />" />
+ */
+export const HeaderTitle = (props: HeaderTitleProps) => {
+  const { className, children, logo, ...rest } = props;
+
+  useTheme();
+  return (
+    <div
+      className={cx('iui-header-logo', className)}
+      role={rest.onClick ? 'button' : undefined}
+      {...rest}
```

Comment:
It needs `tabIndex={0}` and even a keydown handler if you're up for it

Reply:
I can add `tabIndex`, but a keydown handler that does what ?

Reply:
keydown handler to call onClick when space and enter are pressed.

See https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/button_role#required_javascript_features

Reply:
Hmm, what does `tabIndex={0}` supposed to do ? Is it only if onClick is defined ?

Reply:
tabindex=0 makes the element focusable using tab key

Reply:
If adding keydown is too much work, then I would skip tabindex too, because they both go together. I don't know if the app logo necessarily needs keyboard navigation.

@veekeys @bentleyvk Thoughts?

Reply:
Added the keydown handler based on the example provided by the link, added related tests